gcp_variant_transforms/transforms/infer_headers.py

```diff
@@ -242,6 +332,8 @@ def _infer_format_fields(self, variant, defined_headers):
       the field values.
     """
     formats = {}
+    if not self._infer_headers:
```

Comment:
The first time I read this I thought this is dropping all FORMAT fields from the generated header when `infer_headers` is false but `infer_annotation_types` is true. But looking at `vcf_to_bq` module I realized that we keep the originals. While I don't like comments inside code (the code should self describe itself) but this is one of those case that maybe it is a good idea to have a sentence clarifying why it is okay to drop FORMAT fields here.